src/Runner/Linter.php

```diff
@@ -122,11 +137,36 @@ public function run(iterable $files, Ruleset $ruleset, ?FixerInterface $fixer =
             }
             restore_error_handler();
 
-            $rules = $ruleset->getRules();
             foreach ($rules as $rule) {
                 $rule->lintFile($stream, $report, $ignoredViolations);
             }
 
+            if ([] !== $nodeVisitorRules) {
+                $this->setErrorHandler($report, $filePath);
+                try {
+                    $twigSource = new Source($content, $filePath);
+                    $node = $this->env->parse($this->env->tokenize($twigSource));
```

Comment:
If the fixer didn't run, we don't need to parse again the source. We could:
- Write `$node = $this->env->parse($this->env->tokenize($twigSource));` on the `// Validate the file with twig parser.` step
- Reset node ($node = null) only if `if ($newContent !== $content) {` (in the `if (null !== $fixer) {` condition)
- Reparse here the source only if `$node` is null.